awx_collection/plugins/modules/tower_job_launch.py

```diff
@@ -106,6 +118,19 @@
     pass
 
 
+def update_fields(module, p):
+    params = p.copy()
+
+    params_update = {}
+    extra_vars = params.get('extra_vars')
+
+    if extra_vars:
+        params_update['extra_vars'] = [json.dumps(extra_vars)]
```

Comment:
This needs to be shell escaped to actually work. JSON contains characters meaningful to shell and argv parsing.

Reply:
Alternatively, to create a temporary file containing JSON.

Reply:
What do you mean by this exactly? JSON doesn't have many data types, and I'd expect that anything that needs escaping (and conversion back) will be handled by Ansible. Do you have any test cases?

Reply:
I would expect so, too. I'd expect a yaml structure inside `tower_job_launch/extra_vars` to get to tower.

Documenting away a shocking bug isn't solving the bug.

If `tower_job_launch/extra_vars` has a string with an "=" in it, will break this because that is parsed by an argv parser somewhere.

Reply:
I put a variety of non-alphanumeric characters (including `=`) into the `extra_vars` and everything was handled properly:

![Screen Shot 2020-01-08 at 2 06 34 PM](https://user-images.githubusercontent.com/28930622/72008679-5e43cd00-3222-11ea-8911-eb2fe13099da.png)

<img width="364" alt="Screen Shot 2020-01-08 at 2 06 14 PM" src="https://user-images.githubusercontent.com/28930622/72008684-61d75400-3222-11ea-8ae3-cb571b0ef1ad.png">

<img width="331" alt="Screen Shot 2020-01-08 at 2 06 24 PM" src="https://user-images.githubusercontent.com/28930622/72008687-64d24480-3222-11ea-8965-86c063e5217e.png">